### PR TITLE
Use zero underscores for sandboxProfile

### PR DESCRIPTION
**Pull request by pikajude**
@shlevy @nbp

### CODE REVIEW

File: pkgs/os-specific/darwin/apple-sdk/default.nix [hunk 1]
```diff
@@ -97,11 +96,11 @@ let
     propagatedBuildInputs = deps;
 
     # allows building the symlink tree
-    __sandboxProfile = ''
+    sandboxProfile = ''
       (allow file-read* (subpath "/System/Library/Frameworks/${name}.framework"))
     '';
 
-    __propagatedSandboxProfile = stdenv.lib.sandbox.importProfile (generateFrameworkProfile name);
+    __propagatedImpureHostDeps = "/System/Library/Frameworks/${name}.framework/${name}";
```
Comment (shlevy) [inline pkgs/os-specific/darwin/apple-sdk/default.nix]: Have you tested this with new nix yet? Not clear to me how impure host deps interacts with per-drv sandbox profiles

**Comment by shlevy** [comment]
How does this work at all? nix still has `__sandboxProfile`

**Comment by pikajude** [comment]
@shlevy https://github.com/NixOS/nixpkgs/pull/11193/files#diff-3c8a88619c3ca3d61eb0ce8794d9bb74R185

**Comment by shlevy** [comment]
Ah, sorry. Looking over this now.

**Pull request merged**